runner-review.sh

```diff
@@ -0,0 +1,333 @@
+#  Optional flags  :  Number of allowed failed command invocation                                                *
+#                     creating network traffic capture - for failed execution                                    *
+#                     crating system/resources metrics measured during command execution - for failed execution  *
+#                     creating log for all system calls ran by the command - for failed execution                *
+#                     creating command output logs (stdout , stderr) - for failed execution                      *
+#                     Running with debug mode that show each instruction executed by the script                  *
+#                                                                                                                *
+#                                                                                                                *
+#                                                                                                                *
+#  Written by      :  Mohamad Abo-Ras (mohamad.abo.ras@gmail.com)                                                *
+#                                                                                                                *
+#  Date : Jan01,2018                                                                                             *
+#                                                                                                                *
+# Changes History:                                                                                               *
+#                                                                                                                *
+#     Date    |     By     |  Changes/New features                                                               *
+# ----------+------------+----------------------------------------------- *                                      *
+#  01.01.2018 |  Mohamad   | script created for 1st version .             *                                      *
+#           8 |  Abo-Ras   |                                              *                                      *
+# ----------+------------+----------------------------------------------- *                                      *
+#****************************************************************************************************************/
+
+
+
+
+## define global variables
+export DEBUG="set +x"                          ## disabled debug
+export execution_count=0                       ## total executions counter
+export DEFAULT_EXIT_CODE=255                   ## default exit-code
+export EXEC_NAME=$(basename $0)                ## script name 
+export OUTPUT_DIR="`pwd`"                      ## path for output & generated files 
+export TIME_SIGNUTURE="date +"%Y%m%d_%H%M%S""  ## current date&time
```

Comment:
I think that putting this logic in a function makes more sense. 
It's a bit misleading because you have to remember using `$(...)` to evelaute the time signature.

Reply:
This is possible improvement & running . 
All variables exception could be initiated by a function , but the vair
defining a `TIME_SIGNATURE` variable will be defined into the memory with a value of 
and it's called more than one time so it will be more more efficient as the script will access this value from the memory , comparing of calling a function that it will take more time than memory access .